source/slang/slang-ir-legalize-global-values.cpp

```diff
@@ -0,0 +1,231 @@
+#include "slang-ir-legalize-global-values.h"
+
+#include "slang-ir-clone.h"
+#include "slang-ir-util.h"
+
+namespace Slang
+{
+
+void GlobalInstInliningContextGeneric::inlineGlobalValuesAndRemoveIfUnused(IRModule* module)
+{
+    Dictionary<IRInst*, List<IRUse*>> globalInstUsesToInline;
```

Comment:
Why do we need to add everything into a dictionary first?

Reply:
I just changed it to a dictionary so that I can iterate over the keys when deleting later.

Alternatives:
- Use a separate list to record global insts to consider for deletion. These will be the same as the keys in the dictionary.
- Get the global insts from the uses we record, and somehow ensure we're not double-deleting

Reply:
Yeah, just use a List<IRInst*> to store the insts for deletion.

Reply:
Ok I switched to a list. The code also slightly changed meaning (the keys under consideration for deletion may now be a superset of the keys in the dictionary), but AFAIK it should still be ok.